api/eth_transactions.go

```diff
@@ -0,0 +1,508 @@
+package api
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	mathbig "math/big"
+
+	cbg "github.com/whyrusleeping/cbor-gen"
+	"golang.org/x/crypto/sha3"
+
+	"github.com/filecoin-project/go-address"
+	gocrypto "github.com/filecoin-project/go-crypto"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
+	builtintypes "github.com/filecoin-project/go-state-types/builtin"
+	"github.com/filecoin-project/go-state-types/builtin/v10/eam"
+	typescrypto "github.com/filecoin-project/go-state-types/crypto"
+
+	"github.com/filecoin-project/lotus/build"
+	"github.com/filecoin-project/lotus/chain/actors"
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+const Eip1559TxType = 2
+
+type EthTx struct {
+	ChainID              EthUint64   `json:"chainId"`
+	Nonce                EthUint64   `json:"nonce"`
+	Hash                 EthHash     `json:"hash"`
+	BlockHash            EthHash     `json:"blockHash"`
+	BlockNumber          EthUint64   `json:"blockNumber"`
+	TransactionIndex     EthUint64   `json:"transacionIndex"`
+	From                 EthAddress  `json:"from"`
+	To                   *EthAddress `json:"to"`
+	Value                EthBigInt   `json:"value"`
+	Type                 EthUint64   `json:"type"`
+	Input                EthBytes    `json:"input"`
+	Gas                  EthUint64   `json:"gas"`
+	GasLimit             *EthUint64  `json:"gasLimit,omitempty"`
+	MaxFeePerGas         EthBigInt   `json:"maxFeePerGas"`
+	MaxPriorityFeePerGas EthBigInt   `json:"maxPriorityFeePerGas"`
+	V                    EthBytes    `json:"v"`
+	R                    EthBytes    `json:"r"`
+	S                    EthBytes    `json:"s"`
+}
```

Comment:
missing: creates, condition, publicKey, raw,

Reply:
gasPrice is also missing.

Reply:
> missing: creates, condition, publicKey, raw,

These are non-standard fields. I was able to find them in the OpenEthereum docs, but OpenEthereum is Parity's former Eth1 client, which they abandoned. (Yes, it's a misleading name)

A good reference are the official [Eth docs](https://ethereum.org/es/developers/docs/apis/json-rpc/#eth_gettransactionbyhash) (theory), constrating them with the [Infura docs](https://docs.infura.io/infura/networks/ethereum/json-rpc-methods/eth_gettransactionbyhash) (a more practical view).

Reply:
> gasPrice is also missing.

This is an omission, which hasn't bit us yet but it will. So yes, should add. cc @ychiaoli18

Reply:
@raulk It seems that `gasPrice` is only included in the legacy and 2930 transactions but not 1559 transactions ([doc](https://ethereum.github.io/execution-apis/api-documentation/)). Therefore we probably don't need to add this field?